File: java/Easy/IslandPerimeter.java
package Easy;

public class IslandPerimeter {

    public static void main(String[] args){

        int[][] a = {{0, 1, 0, 0}, {1, 1, 1, 0}, {0, 1, 0, 0}, {1, 1, 0, 0}};
        System.out.println(islandPerimeter(a));
    }
    
    public static int islandPerimeter(int[][] grid){

        int p = 0;
        int il = grid.length;
        int jl = grid[0].length;

        for(int i = 0; i < il; i++){
            for(int j = 0; j < jl; j++){
                if(grid[i][j] == 1){
                    p += 4;
                    if(i + 1 < il && grid[i + 1][j] == 1){
                        p -= 2;
                    }
                    if(j + 1 < jl && grid[i][j + 1] == 1){
                        p -= 2;
                    }
                }
            }
        }

        return p;
    }
}

File: python/Easy/IslandPerimeter.py
from typing import List

class Soluton:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        
        p = 0

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    p += 4
                    if i + 1 < len(grid) and grid[i + 1][j] == 1:
                        p -= 2
                    if j + 1 < len(grid[0]) and grid[i][j + 1] == 1:
                        p -= 2

        return p
    
s = Soluton()
a = [[0,1,0,0],[1,1,1,0],[0,1,0,0],[1,1,0,0]]
print(s.islandPerimeter(a))
            

